### PR TITLE
REAL autoplay/pause in view

### PR DESCRIPTION
**Pull request by collin**
@MelizzaP this is the simplest way of doing this.

@spra85 there was discussion of having some sort of smarter video control service that would live on the property. Something like this ?

```js
// if we did something like this there would NOT be handlers for the 'enterviewport' and 'exitviewport' within Revealed
function playOneAndOnlyOneVideo () { /* magic happens */ }
$(document.body).on('enterviewport', 'bulbs-video', playOneAndOnlyOneVideo);
$(document.body).on('exitviewport', 'bulbs-video', playOneAndOnlyOneVideo);
``` 

Using event delegation there: https://learn.jquery.com/events/event-delegation/

## Update
used this branch to make an actual play in view/pause out of view functionality

## Test link
http://ch-rl-pause-vids.test.clickhole.com/video/share-video-your-sisters-wall-let-her-know-youre-n-5249

Videos should pause out of view, and should not play after you manually pause them

### CODE REVIEW

**Comment by MelizzaP** [comment]
@collin @spra85 @kand this is ready for review, still have some safari tests failing in polls. I believe we're going to fix those in 2017

File: elements/bulbs-video/components/revealed.js [hunk 1]
```diff
@@ -258,6 +260,17 @@ export default class Revealed extends React.Component {
       };
     }
 
+    if (typeof this.props.autoplayInView === 'string') {
+      let videoViewport = this.refs.videoViewport;
```
Comment (spra85) [inline elements/bulbs-video/components/revealed.js]: Can we move this block into a separate function?

**Comment by MelizzaP** [comment]
[Video should play when in view, not on load](https://trello.com/c/rHZIANkB/125-video-should-play-when-in-view-not-on-load)

**Comment by collin** [comment]
Yay, so much learning.

File: elements/bulbs-video/components/revealed.js [hunk 2]
```diff
@@ -274,6 +303,16 @@ export default class Revealed extends React.Component {
     }
   }
 
+  handlePauseEvent (reason) {
+    if (reason.pauseReason === 'external') {
+      return;
+    }
+    else if (reason.pauseReason === 'interaction') {
```
Comment (spra85) [inline elements/bulbs-video/components/revealed.js]: Sweet.

Comment (MelizzaP) [inline elements/bulbs-video/components/revealed.js]: yeah, unfortunately this functionality won't work with ads. `jwplayer` does not attach a pause reason to those

Comment (spra85) [inline elements/bulbs-video/components/revealed.js]: Can we make the assumption that the absence of a `pauseReason` implies it is an ad pause or use some kind of player state?

Comment (MelizzaP) [inline elements/bulbs-video/components/revealed.js]: we can assume its an ad if there is no `pauseReason`, in fact `adPause` is its own event.  But we won't be able to determine if scrolling the player out of view paused the ad or if the user paused it.

**Comment by MelizzaP** [comment]
got some slack thumbs

**Pull request merged**